File: 1859-sorting-the-sentence/1859-sorting-the-sentence.cpp
class Solution {
public:
    string sortSentence(string s) {
        map<int,string>m;
        string a="";
        for(int i=0;i<s.size();i++)
        {
            if(isdigit(s[i]))
            {
                int b=(int)s[i];
                m[b]=a;
                cout<<a<<" ";
            }
            else if(isalpha(s[i])) a+=s[i];
            else a="";
        }
        a="";
        for(auto& k:m)
        {
            a+=k.second;
            a+=" ";
        }
        return a.substr(0,a.size()-1);
    }
};

File: 1859-sorting-the-sentence/1859-sorting-the-sentence.py
class Solution:
    def sortSentence(self, s: str) -> str:
        dict={}
        s=list(s.split())
        a=[]
        for i in s:
            dict[i[-1]]=i[0:(len(i)-1)]
        # dict.sort()
        for i in sorted(dict.keys()):
            a.append(dict[i])
        a=' '.join(a)    
        return a